core/deserialization/src/org/jetbrains/kotlin/serialization/deserialization/descriptors/DeserializedClassDescriptor.kt

```diff
@@ -177,7 +177,7 @@ class DeserializedClassDescriptor(
     override fun getInlineClassRepresentation(): InlineClassRepresentation<SimpleType>? = inlineClassRepresentation()
 
     private fun computeInlineClassRepresentation(): InlineClassRepresentation<SimpleType>? {
-        if (!isInlineClass()) return null
+        if (!isInlineOrValueClass()) return null
```

Comment:
BTW even though this change doesn't seem to affect anything, it looks a bit incorrect because multi-field _value_ classes (`LanguageFeature.ValueClasses`, the feature which @zhelenskiy is working on) will not be represented in the way inline classes are represented in Kotlin metadata. We'll need at least a list of all primary constructor properties, not just the single one.

But since we haven't started working on, or even discussing, metadata for multi-field value classes, I don't think it matters a lot.